share/bash-completion/completions/home-manager

```diff
@@ -0,0 +1,552 @@
+# switch
+
+# $ home-manager e<TAB>
+#
+# edit
+# expire-generations
+
+# $ home-manager build -A programs.e<TAB>
+#
+# programs.eclipse
+# programs.eclipse.plugins  
+# programs.emacs
+# programs.emacs.overrides
+# programs.emacs.package    
+
+# $ home-manager remove-generations 20<TAB>
+#
+# 200
+# 201
+# 202
+# 203
+
+##################################################
+# Notes:
+
+# « home-manager » Subcommands:
+#
+#   help
+#   edit
+#   build
+#   switch
+#   generations
+#   remove-generations
+#   expire-generations
+#   packages
+#   news
+
+# « home-manager » Options:
+#
+#   -f FILE
+#   -A ATTRIBUTE
+#   -I PATH
+#   -v
+#   -n
+#   -h
+
+# $ home-manager
+# 
+# Usage: /home/sboo/.nix-profile/bin/home-manager [OPTION] COMMAND
+# 
+# Options
+# 
+#   -f FILE      The home configuration file.
+#                Default is '~/.config/nixpkgs/home.nix'.
+#   -A ATTRIBUTE Optional attribute that selects a configuration
+#                expression in the configuration file.
+#   -I PATH      Add a path to the Nix expression search path.
+#   -v           Verbose output
+#   -n           Do a dry run, only prints what actions would be taken
+#   -h           Print this help
+# 
+# Commands
+# 
+#   help         Print this help
+# 
+#   edit         Open the home configuration in $EDITOR
+# 
+#   build        Build configuration into result directory
+# 
+#   switch       Build and activate configuration
+# 
+#   generations  List all home environment generations
+# 
+#   remove-generations ID...
+#       Remove indicated generations. Use 'generations' command to
+#       find suitable generation numbers.
+# 
+#   expire-generations TIMESTAMP
+#       Remove generations older than TIMESTAMP where TIMESTAMP is
+#       interpreted as in the -d argument of the date tool. For
+#       example "-30 days" or "2018-01-01".
+# 
+#   packages     List all packages installed in home-manager-path
+# 
+#   news         Show news entries in a pager
+# 
+
+##################################################
+# Dependencies:
+
+command -v home-manager >/dev/null
+command -v grep         >/dev/null
+command -v sed          >/dev/null
+
+##################################################
+# Code:
+
+_home-manager_list_generation_identifiers ()
+
+{
+
+    home-manager generations  |  sed -n -e 's/^................ : id \([[:alnum:]]\+\) -> .*/\1/p'
+
+}
+
+# NOTES
+#
+# (1) the « sed -n -e 's/.../.../p' » invocation:
+#
+#    * the « -e '...' » option takes a Sed Script.
+#    * the « -n » option only prints when « .../p » would print.
+#    * the « s/xxx/yyy/ » Sed Script substitutes « yyy » whenever « xxx » is matched.
+#
+# (2) the « '^................ : id \([[:alnum:]]\+\) -> .*' » regular expression:
+#
+#    * matches « 199 », for example, in the line « 2019-03-13 15:26 : id 199 -> /nix/store/mv619y9pzgsx3kndq0q7fjfvbqqdy5k8-home-manager-generation »
+#
+# 
+
+#------------------------------------------------#
+
+_home-manager_list_nix_attributes ()
```

Comment:
These values don't really reflect the intended functionality of the `-A` option. The option is there to allow a `home.nix` of the form
```nix
{
  setup-a = import ./home-a.nix;
  setup-b = import ./home-b.nix;
}
```
where `home-a.nix` and `home-b.nix` are two separate Home Manager configurations.

You can then choose which one to activate using
```console
$ home-manager -A setup-a switch
```
and
```console
$ home-manager -A setup-b switch
```

This is not a super well advertised feature. I might write some proper documentation for it at some point 🙂

Reply:
thanks the reply!

(wnd thanks for the correction
 yeah, I just guessed what it did; trying `programs.emacs` worked, so I assumed it solved the problem I had of quickly testing susbets of my configuration. the actual intent makes more sense, and seems very useful! I think I'll use it to implement a "minimal" config to "maximal" one.)

Reply:
Maybe it would be a good idea to json-instantiate the top-level home-manager record and then show the keys as values, if that is how "-A" should be used?

Reply:
@leotaku (btw, if you have a Mac, could you test the bash script?)